src/Logic.hs

```diff
@@ -228,17 +228,21 @@ handlePullRequestCommitChanged pr newSha state =
     closedState = handlePullRequestClosed pr state
     update pullRequest =
       let
-        oldSha   = Pr.sha pullRequest
-        branch   = Pr.branch pullRequest
-        title    = Pr.title pullRequest
-        author   = Pr.author pullRequest
-        newState = closedState >>= handlePullRequestOpened pr branch newSha title author
+        branch = Pr.branch pullRequest
+        title  = Pr.title pullRequest
+        author = Pr.author pullRequest
       in
-        -- If the change notification was a false positive, ignore it.
-        if oldSha == newSha then return state else newState
+        closedState >>= handlePullRequestOpened pr branch newSha title author
   in
-    -- If the pull request was not present in the first place, do nothing.
-    maybe (return state) update $ Pr.lookupPullRequest pr state
+    case Pr.lookupPullRequest pr state of
+      -- If the change notification was a false positive, ignore it.
+      Just pullRequest | Pr.sha pullRequest == newSha -> pure state
+      -- If the new commit hash is one that we pushed ourselves, ignore the
+      -- change too, we don't want to lose the approval status.
+      Just pullRequest | newSha `Pr.wasIntegrationCandidateFor` pullRequest -> pure state
```

Comment:
Possible bug: what happens when someone (force) pushes a commit that was previously approved/the integration candidate?

Reply:
Then we ignore that, but it is not a problem, because if it was an integration candidate, then it was good to be merged, and if it no longer is an integration candidate, then we failed to merge it so it is no longer relevant.

Reply:
Thanks for clarifying!

It might be good to test this behavior, but I'll leave it up to you to judge whether that's worth it.

Reply:
That test is essentially the one I added in this pull request. What we know from GitHub is that the commit changed, but not who caused that change.